packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/LinkToCategoriesTests.java

```diff
@@ -0,0 +1,276 @@
+
+        STEP("Verify if both categories are present in file metadata");
+        fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+
+        fileNode.assertThat().field(ASPECTS_FIELD).contains("cm:generalclassifiable");
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains("cm:categories");
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(category.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(secondCategory.getId());
+    }
+
+    /**
+     * Link content, which already has some linked category to new ones and verify if all categories are present in node's properties
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_usingContentWithAlreadyLinkedCategories()
+    {
+        STEP("Link content to created category");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+        restClient.assertStatusCodeIs(CREATED);
+
+        STEP("Create second and third category under root, link content to them and expect 201");
+        final RestCategoryModel secondCategory = prepareCategoryUnderRoot();
+        final RestCategoryModel thirdCategory = prepareCategoryUnderRoot();
+        final List<RestCategoryLinkBodyModel> categoryLinks = List.of(
+            createCategoryLinkWithId(secondCategory.getId()),
+            createCategoryLinkWithId(thirdCategory.getId())
+        );
+        final RestCategoryModelsCollection linkedCategories = restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategories(categoryLinks);
+
+        restClient.assertStatusCodeIs(CREATED);
+        linkedCategories.assertThat().entriesListCountIs(2);
+        linkedCategories.getEntries().get(0).onModel().assertThat().isEqualTo(secondCategory);
+        linkedCategories.getEntries().get(1).onModel().assertThat().isEqualTo(thirdCategory);
+
+        STEP("Verify if all three categories are present in file metadata");
+        final RestNodeModel fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains("cm:categories");
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(category.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(secondCategory.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(thirdCategory.getId());
+    }
+
+    /**
+     * Try to link content to category as an improper user and expect 403 (Forbidden)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_usingUserWithoutAccessRightsAndExpect403()
+    {
+        STEP("Try to link content to a category using user without read permission and expect 403");
```

Comment:
I think we should try a user without change permission instead (and expect 403).

Reply:
I don't know why but I assumed that read permission will be enough to create a link. Now after rethinking you are right - I will fix it.